test/uk/gov/hmrc/excisemovementcontrolsystemapi/services/MessageServiceSpec.scala

```diff
@@ -768,6 +768,107 @@ class MessageServiceSpec extends PlaySpec
         verify(movementRepository).save(ern2Movement)
       }
     }
+
+    "there are many messages" should {
+      "update them if they all correspond to one movement" in {
+        val message1 = IE801Message.createFromXml(XmlMessageGeneratorFactory.generate("ern1", MessageParams(IE801, "XI0000021a", Some("AT00000602078"), Some("arc"), Some("lrn1"))))
+        val message2 = IE819Message.createFromXml(XmlMessageGeneratorFactory.generate("ern1", MessageParams(IE819, "X00008a", Some("token"), Some("arc"))))
+        val message3 = IE807Message.createFromXml(XmlMessageGeneratorFactory.generate("ern1", MessageParams(IE807, "XI0000021b", Some("AT00000602078"), Some("arc"), Some("lrn1"))))
+        val message4 = IE840Message.createFromXml(XmlMessageGeneratorFactory.generate("ern1", MessageParams(IE840, "X00008b", Some("token"), Some("arc"))))
+
+        val ern1NewMessages = Seq(message1, message2, message3, message4)
+        val ern1Movement = Movement(
+          None,
+          "lrn1",
+          "ern1",
+          None,
+          messages = Seq.empty
+        )
+
+        when(dateTimeService.timestamp()).thenReturn(now)
+        when(movementRepository.getAllBy(eqTo("ern1"))).thenReturn(Future.successful(Seq(ern1Movement)))
+        when(movementRepository.save(any)).thenReturn(Future.successful(Done))
+        when(ernRetrievalRepository.getLastRetrieved(any)).thenReturn(Future.successful(None))
+        when(boxIdRepository.getBoxIds(any)).thenReturn(Future.successful(Set.empty))
+        when(messageConnector.getNewMessages(eqTo("ern1"))(any)).thenReturn(Future.successful(GetMessagesResponse(ern1NewMessages, 1)))
+        when(messageConnector.acknowledgeMessages(any)(any)).thenReturn(Future.successful(Done))
+
+        messageService.updateAllMessages(Set("ern1")).futureValue
```

Comment:
This method was setup to allow multiple ERNs to be passed into `updateMessages` but handle any errors for an ERN in a common way instead of having to replicate the error handling in all places we wanted to call updateMessages. This test is technically for the `updateMessages` method and you are only passing in one ERN. Not a big deal but maybe we should just call `updateMessages` directly for this test?